### PR TITLE
Add resize image for VLM benchmarking

### PR DESCRIPTION
**Pull request by JohnLeFeng**
## Description

This PR mainly add image resize function for VLM benchmark sample to process images with different image size as different image size might bring different image embedding size and then makes performance different but bring same output result.

Below is detail of changes
* Add resize image function
* Add option for resize image in VLM benchmark
* Show input token in benchmarking output
* Update Readme with performance data when resized image
* Remove 2-character short option from C++ sample readme due to cxxopts only support 1-character short option
* Add test to run VLM benchmark with different image size
* Add test to run VLM benchmark with negative image size

[CVS-185186](https://jira.devtools.intel.com/browse/CVS-185186)

## Checklist:
- [x] This PR follows [GenAI Contributing guidelines](https://github.com/openvinotoolkit/openvino.genai?tab=contributing-ov-file#contributing). 
- [x] Tests have been updated or added to cover the new code. 
- [x] This PR fully addresses the ticket. 
- [x] I have made corresponding changes to the documentation.

### CODE REVIEW

File: samples/python/visual_language_chat/README.md [hunk 1]
```diff
@@ -108,30 +108,55 @@ python benchmark_vlm.py [OPTIONS]
...
-- `-rw, --relevance_weight` (optional): Float value from 0 to 1, control the trade-off between diversity and relevance for visual tokens pruning, a value of 0 disables relevance weighting, while higher values (up to 1.0) emphasize relevance, making pruning more conservative on borderline tokens.
+- `-pr, --pruning_ratio` (optional): Percentage of visual tokens to prune (valid range: 0-100). If this option is not provided, pruning is disabled.
+- `-rw, --relevance_weight` (optional): Float value from 0 to 1, controls the trade-off between diversity and relevance for visual tokens pruning; a value of 0 disables relevance weighting, while higher values (up to 1.0) emphasize relevance, making pruning more conservative on borderline tokens.
 
 ### Output:
 
 ```
-python benchmark_vlm.py -m miniCPM-V-2_6 -i 319483352-d5fbbd1a-d484-415c-88cb-9986625b7b11.jpg -n 3
+python benchmark_vlm.py -m miniCPM-V-2_6 -i 319483352-d5fbbd1a-d484-415c-88cb-9986625b7b11.jpg -n 3 -d GPU
 ```
```
[the remark this answers is not in the document]
Comment (JohnLeFeng) [reply samples/python/visual_language_chat/README.md:124]: this should be fix.

**Review by rkazants — CHANGES_REQUESTED** [review]
Please refer JIRA ticket in PR description. why this feature is needed.

File: samples/cpp/visual_language_chat/benchmark_vlm.cpp [hunk 2]
```diff
@@ -59,7 +61,11 @@ int main(int argc, char* argv[]) try {
     std::string device = result["device"].as<std::string>();
     size_t num_warmup = result["num_warmup"].as<size_t>();
     size_t num_iter = result["num_iter"].as<size_t>();
-    std::vector<ov::Tensor> images = utils::load_images(image_path);
+    int image_height = result["image_height"].as<int>();
+    int image_width = result["image_width"].as<int>();
```
Comment (rkazants) [inline samples/cpp/visual_language_chat/benchmark_vlm.cpp:65]: don't use `int`. Instead, we need to use `int32_t` or `int64_t`, explicit types

File: samples/cpp/visual_language_chat/README.md [hunk 3]
```diff
@@ -85,32 +85,57 @@ benchmark_vlm [OPTIONS]
 
 - `-m, --model`(default: `.`): Path to the model and tokenizers base directory.
 - `-p, --prompt` (default: ''): The prompt to generate text. If without `-p` and `--pf`, the default prompt is `"What is on the image?"`
-- `--pf, --prompt_file` Read prompt from file.
+- `--prompt_file` Read prompt from file.
 - `-i, --image` (default: `image.jpg`): Path to the image.
-- `-nw, --num_warmup` (default: `1`): Number of warmup iterations.
-- `-mt, --max_new_tokens` (default: `20`): Maximal number of new tokens.
+- `-H, --image_height` (default: `0`): Target image height (if resizing is needed).
+- `-W, --image_width` (default: `0`): Target image width (if resizing is needed).
```
Comment (rkazants) [inline samples/cpp/visual_language_chat/README.md:91]: weird default values, please double-check

**Comment by JohnLeFeng** [comment]
> Please refer JIRA ticket in PR description. why this feature is needed.

Just create the ticket. [CVS-185186](https://jira.devtools.intel.com/browse/CVS-185186)

File: samples/python/visual_language_chat/README.md [hunk 4]
```diff
@@ -108,30 +108,55 @@ python benchmark_vlm.py [OPTIONS]
...
 - `-mt, --max_new_tokens` (default: `20`): Maximal number of new tokens.
-- `-n, --num_iter` (default: `3`): Number of iterations.
+- `-n, --num_iter` (default: `2`): Number of iterations.
 - `-d, --device` (default: `"CPU"`): Device to run the model on.
-- `-pr, --pruning_ratio`: (optional): Percentage of visual tokens to prune (valid range: 0-100). If this option is not provided, pruning is disabled.
-- `-rw, --relevance_weight` (optional): Float value from 0 to 1, control the trade-off between diversity and relevance for visual tokens pruning, a value of 0 disables relevance weighting, while higher values (up to 1.0) emphasize relevance, making pruning more conservative on borderline tokens.
+- `-pr, --pruning_ratio` (optional): Percentage of visual tokens to prune (valid range: 0-100). If this option is not provided, pruning is disabled.
+- `-rw, --relevance_weight` (optional): Float value from 0 to 1, controls the trade-off between diversity and relevance for visual tokens pruning; a value of 0 disables relevance weighting, while higher values (up to 1.0) emphasize relevance, making pruning more conservative on borderline tokens.
```
[the remark this answers is not in the document]
Comment (JohnLeFeng) [reply samples/python/visual_language_chat/README.md]: thanks, updated it.

File: samples/python/visual_language_chat/benchmark_vlm.py [hunk 5]
```diff
@@ -8,28 +8,25 @@
...
-        path: The path to the image.
-
-    Returns: the ov.Tensor containing the image.
-
-    '''
+def read_image(path: str, target_height: Optional[int] = None, target_width: Optional[int] = None) -> Tensor:
     pic = Image.open(path).convert("RGB")
+    if (target_height is None) != (target_width is None):
+        raise ValueError("target_height and target_width must be provided together")
+    if target_height is not None and target_width is not None:
```
[the remark this answers is not in the document]
Comment (JohnLeFeng) [reply samples/python/visual_language_chat/benchmark_vlm.py]: This is solved.

File: samples/python/visual_language_chat/benchmark_vlm.py [hunk 6]
```diff
@@ -8,28 +8,25 @@
...
+        pic = pic.resize((target_width, target_height))
```
[the remark this answers is not in the document]
Comment (JohnLeFeng) [reply samples/python/visual_language_chat/benchmark_vlm.py]: This is not required; Main purpose of this PR is showing the different image size brings different token length.

File: samples/cpp/visual_language_chat/load_image.hpp [hunk 7]
```diff
@@ -5,9 +5,11 @@
...
+#include <cstdint>
+#include <optional>
 #include <filesystem>
 
 namespace utils {
-ov::Tensor load_image(const std::filesystem::path& image_path);
-std::vector<ov::Tensor> load_images(const std::filesystem::path& image_path);
+ov::Tensor load_image(const std::filesystem::path& image_path, std::optional<int32_t> target_height = std::nullopt, std::optional<int32_t> target_width = std::nullopt);
+std::vector<ov::Tensor> load_images(const std::filesystem::path& image_path, std::optional<int32_t> target_height = std::nullopt, std::optional<int32_t> target_width = std::nullopt);
 }
```
[the remark this answers is not in the document]
Comment (JohnLeFeng) [reply samples/cpp/visual_language_chat/load_image.hpp:20]: This is fine due to original implementation is using `std::vector` already.

File: samples/cpp/visual_language_chat/load_image.cpp [hunk 8]
```diff
@@ -6,26 +6,35 @@
...
+std::vector<ov::Tensor> utils::load_images(const std::filesystem::path& input_path, std::optional<int32_t> target_height, std::optional<int32_t> target_width) {
     if (input_path.empty() || !fs::exists(input_path)) {
         throw std::runtime_error{"Path to images is empty or does not exist."};
     }
     if (fs::is_directory(input_path)) {
         std::set<fs::path> sorted_images{fs::directory_iterator(input_path), fs::directory_iterator()};
         std::vector<ov::Tensor> images;
         for (const fs::path& dir_entry : sorted_images) {
-            images.push_back(utils::load_image(dir_entry));
+            images.push_back(utils::load_image(dir_entry, target_height, target_width));
```
[the remark this answers is not in the document]
Comment (JohnLeFeng) [reply samples/cpp/visual_language_chat/load_image.cpp]: This is fine due to original implementation is using std::vector already.

File: samples/cpp/visual_language_chat/load_image.cpp [hunk 9]
```diff
@@ -6,26 +6,35 @@
...
 #define STB_IMAGE_IMPLEMENTATION
 #include "stb_image.h"
+#define STB_IMAGE_RESIZE_IMPLEMENTATION
+#include "stb_image_resize2.h"
 #include "load_image.hpp"
 
 namespace fs = std::filesystem;
 
-std::vector<ov::Tensor> utils::load_images(const std::filesystem::path& input_path) {
+std::vector<ov::Tensor> utils::load_images(const std::filesystem::path& input_path, std::optional<int32_t> target_height, std::optional<int32_t> target_width) {
```
Comment (yatarkan) [inline samples/cpp/visual_language_chat/load_image.cpp]: Let's use `std::optional<std::pair>` or `std::optional<ImageSize>`.
The same for `load_image` function.

File: samples/python/visual_language_chat/benchmark_vlm.py [hunk 10]
```diff
@@ -8,28 +8,25 @@
...
-def read_image(path: str) -> Tensor:
-    '''
-
-    Args:
-        path: The path to the image.
-
-    Returns: the ov.Tensor containing the image.
-
-    '''
+def read_image(path: str, target_height: Optional[int] = None, target_width: Optional[int] = None) -> Tensor:
```
Comment (yatarkan) [inline samples/python/visual_language_chat/benchmark_vlm.py]: Let's use `Optional[tuple[int, int]]`

File: samples/cpp/visual_language_chat/benchmark_vlm.cpp
Comment (yatarkan) [inline samples/cpp/visual_language_chat/benchmark_vlm.cpp]: If only single character shortcut is supported, let's remove invalid ones

File: samples/python/visual_language_chat/benchmark_vlm.py [hunk 11]
```diff
@@ -52,17 +49,25 @@ def main():
     parser.add_argument("-p", "--prompt", type=str, default=None, help="Prompt")
     parser.add_argument("-pf", "--prompt_file", type=str, help="Read prompt from file")
     parser.add_argument("-i", "--image", type=str, default="image.jpg", help="Image")
+    parser.add_argument(
+        "-ih", "--image_height", type=int, default=None, help="Target image height (if resizing is needed)"
+    )
+    parser.add_argument(
+        "-iw", "--image_width", type=int, default=None, help="Target image width (if resizing is needed)"
+    )
```
Comment (yatarkan) [inline samples/python/visual_language_chat/benchmark_vlm.py:75]: Let's align with C++ sample:

```suggestion
    parser.add_argument(
        "-H", "--image_height", type=int, default=None, help="Target image height (if resizing is needed)"
    )
    parser.add_argument(
        "-W", "--image_width", type=int, default=None, help="Target image width (if resizing is needed)"
    )
```

And update README for python sample.

File: samples/cpp/visual_language_chat/benchmark_vlm.cpp [hunk 12]
```diff
@@ -86,7 +95,7 @@ int main(int argc, char* argv[]) try {
 
     auto input_data = pipe->get_tokenizer().encode(prompt);
     size_t prompt_token_size = input_data.input_ids.get_shape()[1];
-    std::cout << "Number of images:" << images.size() << ", prompt token size:" << prompt_token_size << std::endl;
+    std::cout << "Number of images: " << images.size() << ", prompt token size: " << prompt_token_size << std::endl;
```
Comment (yatarkan) [inline samples/cpp/visual_language_chat/benchmark_vlm.cpp]: I would also print if resizing is applied to images (and print target size). The same in python sample.

File: tests/python_tests/samples/test_benchmark_vlm.py [hunk 13]
```diff
@@ -29,3 +30,91 @@ def test_sample_benchmark_vlm(self, convert_model, download_test_content):
...
+            "--image_height",
+            image_height,
+            "--image_width",
+            image_width,
+        ]
+        run_sample(benchmark_py_command)
+
+    @pytest.mark.vlm
+    @pytest.mark.samples
+    def test_sample_benchmark_vlm_negative_dimensions(self):
```
Comment (yatarkan) [inline tests/python_tests/samples/test_benchmark_vlm.py]: Let's extend this test with and test all negative scenarios in it - negative dimensions + single dimension (if only height or width provided)

```suggestion
    def test_sample_benchmark_vlm_fails(self):
```

File: tests/python_tests/samples/test_benchmark_vlm.py [hunk 14]
```diff
@@ -29,3 +30,91 @@ def test_sample_benchmark_vlm(self, convert_model, download_test_content):
...
+        # Validation of negative dimensions in the Python script raises RuntimeError
+        # before any image or model is loaded, so no fixtures are required.
```
[the remark this answers is not in the document]
Comment (yatarkan) [reply tests/python_tests/samples/test_benchmark_vlm.py]: Consider Copilot's suggestion

File: samples/cpp/visual_language_chat/CMakeLists.txt [hunk 15]
```diff
@@ -17,6 +17,11 @@ file(DOWNLOAD
     ${CMAKE_BINARY_DIR}/stb_image.h
     EXPECTED_HASH MD5=27932e6fb3a2f26aee2fc33f2cb4e696)
 
+file(DOWNLOAD
+    https://raw.githubusercontent.com/nothings/stb/f75e8d1cad7d90d72ef7a4661f1b994ef78b4e31/stb_image_resize2.h
+    ${CMAKE_BINARY_DIR}/stb_image_resize2.h
+    EXPECTED_HASH MD5=9293a158ddf9e44dd186c446372784d8)
```
[the remark this answers is not in the document]
Comment (JohnLeFeng) [reply samples/cpp/visual_language_chat/CMakeLists.txt:23]: I think we can keep this, it aligns with `stb_image.h`.

File: samples/cpp/visual_language_chat/load_image.cpp [hunk 16]
```diff
@@ -36,6 +45,27 @@ ov::Tensor utils::load_image(const std::filesystem::path& image_path) {
...
+        unsigned char* resized = static_cast<unsigned char*>(
+            malloc(size_t(*target_width) * size_t(*target_height) * desired_channels));
+        if (!resized) {
+            stbi_image_free(image);
+            throw std::runtime_error{"Failed to allocate memory for resized image."};
+        }
+        unsigned char* resize_result = stbir_resize_uint8_linear(
+            image, x, y, 0,
+            resized, *target_width, *target_height, 0,
+            static_cast<stbir_pixel_layout>(desired_channels));
```
[the remark this answers is not in the document]
Comment (JohnLeFeng) [reply samples/cpp/visual_language_chat/load_image.cpp:55]: I think that is fine in current sample due to vision embedding only processes 3 channels image.

File: tests/python_tests/samples/test_benchmark_vlm.py [hunk 17]
```diff
@@ -29,3 +30,166 @@ def test_sample_benchmark_vlm(self, convert_model, download_test_content):
...
+                    "-m",
+                    "fake_model",
+                    "-i",
+                    "fake_image.jpg",
+                    "-H",
+                    "0",
+                    "-W",
+                    "0",
+                ]
+            )
```
Comment (yatarkan) [inline tests/python_tests/samples/test_benchmark_vlm.py]:
```suggestion
    def test_sample_benchmark_vlm_invalid_resize(self):
        # Validation of invalid resize dimensions causes the Python sample subprocess to fail
        # before any image or model is loaded, so this test observes CalledProcessError.
        benchmark_script = SAMPLES_PY_DIR / "visual_language_chat/benchmark_vlm.py"
        
        invalid_sizes = [
            (-1, 224),
            (224, -1),
            (0, 224),
            (224, 0),
            (0, 0),
            (None, 224),
            (224, None)
        ]
        
        for height, width in invalid_sizes:
            command = [sys.executable, benchmark_script, "-m", "fake_model", "-i", "fake_image.jpg",]
            
            if height is not None:
                command.extend(["-H", height])
            
            if width is not None:
                command.extend(["-W", width])

            with pytest.raises(subprocess.CalledProcessError):
                run_sample(command)
```

**Review by yatarkan — COMMENTED** [review]
Other changes LGTM

File: tests/python_tests/samples/test_benchmark_vlm.py (hunk 17 above)
Comment (JohnLeFeng) [reply tests/python_tests/samples/test_benchmark_vlm.py]: Thanks, this is good.

**Review by yatarkan — COMMENTED** [review]
Please apply Copilot's comments as well

File: samples/cpp/visual_language_chat/benchmark_vlm.cpp [hunk 18]
```diff
@@ -59,7 +63,17 @@ int main(int argc, char* argv[]) try {
...
+
+    if (!(result.count("image_height") && result.count("image_width"))) {
+        std::cout << "image_height and image_width must be provided together!" << std::endl;
+        return EXIT_FAILURE;
+    }
+    const std::optional<utils::ImageSize> image_size = 
+        (result.count("image_width") && result.count("image_height"))
+        ? std::optional<utils::ImageSize>{utils::ImageSize{result["image_width"].as<int32_t>(), result["image_height"].as<int32_t>()}}
+        : std::nullopt;
+
```
[the remark this answers is not in the document]
Comment (JohnLeFeng) [reply samples/cpp/visual_language_chat/benchmark_vlm.cpp]: Thanks for the review on line `67`, but I think line `71` to `75` is fine due to load_image will check whether the `image_width` and `image_height` are larger than `0`.

File: tests/python_tests/samples/test_benchmark_vlm.py [hunk 19]
```diff
@@ -29,3 +30,81 @@ def test_sample_benchmark_vlm(self, convert_model, download_test_content):
...
+        ]
+        
+        for height, width in invalid_sizes:
+            command = [sys.executable, benchmark_script, "-m", "fake_model", "-i", "fake_image.jpg",]
+            
+            if height is not None:
+                command.extend(["-H", height])
+            
+            if width is not None:
+                command.extend(["-W", width])
```
[the remark this answers is not in the document]
Comment (JohnLeFeng) [reply tests/python_tests/samples/test_benchmark_vlm.py]: thanks

File: samples/cpp/visual_language_chat/README.md [hunk 20]
```diff
@@ -84,33 +84,61 @@ benchmark_vlm [OPTIONS]
...
+	Input token size: 80
+	Output token size: 20
+	Load time: 4460.00 ms
+	Generate time: 323.61 ± 8.40 ms
+	Tokenization time: 15.80 ± 0.05 ms
+	Detokenization time: 0.24 ± 0.01 ms
+	Embeddings preparation time: 36.70 ± 0.00 ms
+	TTFT: 72.10 ± 4.37 ms
+	TPOT: 13.21 ± 3.34 ms
+	Throughput: 75.69 ± 19.10 tokens/s
```
[the remark this answers is not in the document]
Comment (JohnLeFeng) [reply samples/cpp/visual_language_chat/README.md:139]: thanks, but indent is incorrect.

File: tests/python_tests/samples/test_benchmark_vlm.py [hunk 21]
```diff
@@ -29,3 +30,85 @@ def test_sample_benchmark_vlm(self, convert_model, download_test_content):
...
+        for height, width in invalid_sizes:
+            cpp_command = [benchmark_sample, "-m", "fake_model", "-i", "fake_image.jpg"]
+            py_command = [
+                sys.executable,
+                benchmark_script,
+                "-m",
+                "fake_model",
+                "-i",
+                "fake_image.jpg",
+            ]
```
[the remark this answers is not in the document]
Comment (JohnLeFeng) [reply tests/python_tests/samples/test_benchmark_vlm.py:107]: Updated, but keep `fake_image.jpg` in python command.

File: samples/cpp/visual_language_chat/benchmark_vlm.cpp [hunk 22]
```diff
@@ -2,6 +2,8 @@
 // SPDX-License-Identifier: Apache-2.0
 
 #include <cxxopts.hpp>
+#include <cstdint>
+#include <optional>
 #include <filesystem>
```
[the remark this answers is not in the document]
Comment (JohnLeFeng) [reply samples/cpp/visual_language_chat/benchmark_vlm.cpp:8]: I think this can be updated when changes include trees.

File: samples/cpp/visual_language_chat/benchmark_vlm.cpp [hunk 23]
```diff
@@ -54,12 +58,29 @@ int main(int argc, char* argv[]) try {
...
     } 
 
-    const std::string models_path = result["model"].as<std::string>();
+    std::string models_path;
+    if (result.count("model")) {
+        models_path = result["model"].as<std::string>();
+    } else {
+        std::cout << "Model path is not provided!" << std::endl;
+        return EXIT_FAILURE;
+    }
```
[the remark this answers is not in the document]
Comment (JohnLeFeng) [reply samples/cpp/visual_language_chat/benchmark_vlm.cpp:67]: No, it will not count due to argument `model` doesn't provide.

File: samples/python/visual_language_chat/README.md [hunk 24]
```diff
@@ -104,34 +104,60 @@ python benchmark_vlm.py [OPTIONS]
...
-- `-m, --model`(default: `.`): Path to the model and tokenizers base directory.
-- `-p, --prompt` (default: `None`): The prompt to generate text. If without `-p` and `-pf`, the default prompt is `"What is on the image?"`
-- `-pf, --prompt_file` Read prompt from file.
-- `-i, --image` (default: `image.jpg`): Path to the image.
-- `-nw, --num_warmup` (default: `1`): Number of warmup iterations.
-- `-mt, --max_new_tokens` (default: `20`): Maximal number of new tokens.
-- `-n, --num_iter` (default: `3`): Number of iterations.
+- `-m, --model` (required): Path to the model and tokenizers base directory.
+- `-p, --prompt` (default: `None`): The prompt to generate text. If without `-p, --prompt`, and `-F, --prompt_file`, the default prompt is `"What is on the image?"`
+- `-F, --prompt_file`: Read prompt from file.
```
[the remark this answers is not in the document]
Comment (JohnLeFeng) [reply samples/python/visual_language_chat/README.md:109]: No, document isn't including this, so no need to add.

**Comment by yatarkan** [comment]
build_jenkins

**Review by yatarkan — APPROVED** [review]

File: samples/cpp/visual_language_chat/README.md [hunk 25]
```diff
@@ -83,34 +83,62 @@ benchmark_vlm [OPTIONS]
...
 
-- `-m, --model`(default: `.`): Path to the model and tokenizers base directory.
-- `-p, --prompt` (default: ''): The prompt to generate text. If without `-p` and `--pf`, the default prompt is `"What is on the image?"`
-- `--pf, --prompt_file` Read prompt from file.
-- `-i, --image` (default: `image.jpg`): Path to the image.
-- `-nw, --num_warmup` (default: `1`): Number of warmup iterations.
-- `-mt, --max_new_tokens` (default: `20`): Maximal number of new tokens.
+- `-m, --model` (default: ''): Path to the model and tokenizers base directory.
+- `-p, --prompt` (default: ''): The prompt to generate text. If without `-p, --prompt` and `-F, --prompt_file`, the default prompt is `"What is on the image?"`
+- `-F, --prompt_file` Read prompt from file.
```
Comment (rkazants) [inline samples/cpp/visual_language_chat/README.md:88]:
```suggestion
- `--pf, --prompt_file` Read prompt from file.
```
why changed? This PR is dedicated to different functionality.

File: samples/cpp/visual_language_chat/README.md [hunk 26]
```diff
@@ -83,34 +83,62 @@ benchmark_vlm [OPTIONS]
...
+- `-N, --num_warmup` (default: `1`): Number of warmup iterations.
 - `-n, --num_iter` (default: `3`): Number of iterations.
+- `-M, --max_new_tokens` (default: `20`): Maximal number of new tokens.
 - `-d, --device` (default: `"CPU"`): Device to run the model on.
-- `-pr, --pruning_ratio`: (optional): Percentage of visual tokens to prune (valid range: 0-100); if this option is not provided, pruning is disabled.
-- `-rw, --relevance_weight` (optional): Float value from 0 to 1, controls the trade-off between diversity and relevance for visual tokens pruning; a value of 0 disables relevance weighting, while higher values (up to 1.0) emphasize relevance, making pruning more conservative on borderline tokens.
+- `-P, --pruning_ratio`: (optional): Percentage of visual tokens to prune (valid range: 0-100); if this option is not provided, pruning is disabled.
+- `-R, --relevance_weight` (optional): Float value from 0 to 1, controls the trade-off between diversity and relevance for visual tokens pruning; a value of 0 disables relevance weighting, while higher values (up to 1.0) emphasize relevance, making pruning more conservative on borderline tokens.
```
Comment (rkazants) [inline samples/cpp/visual_language_chat/README.md:97]: why?

File: samples/cpp/visual_language_chat/README.md (hunk 25 above)
Comment (JohnLeFeng) [reply samples/cpp/visual_language_chat/README.md:88]: Hi @rkazants, the reason to change this is cxxopts only support 1-character short option. And this PR is using single capital for image size, height(H) and width(W). (avoid to use “h”) Therefore, we changed all 2-character short option to 1-character capital to align changes for image size option.

Comment (rkazants) [reply samples/cpp/visual_language_chat/README.md:88]: you change API tool that is not accepted. We may impact users who rely on this option.
I think you mix up different functionality: cxxopts, new options for resize. Let us have here only new options for resize. If you want to adjust option names, let us do it separately.

**Review by rkazants — APPROVED** [review]
agreed to accept
